### DISCUSSION TITLE
Standard API controllers along side RpcControllers

### DISCUSSION THREAD

**Discussion by eric-davis** [edjCase/JsonRpc#122] [Q&A] [answered]
Is having standard API controllers along side RpcControllers supported? I have a standard ASP.NET API controller defined with a GET endpoint. And I also have a RpcController with 1 method defined. The standard controller was created first and was working as expected, but once I added the RpcController (using the AddController<TController>() extension method) I stared getting a -32600 RPC error ("Json request was empty") when invoking the original standard GET endpoint.

**Comment by Gekctek** [comment]
@eric-davis Could you show your `public void Configure(IApplicationBuilder app)` in your Startup.cs file so I can see your setup.
There should not be anything preventing them running side by side, there might just need a tweak in your setup.

**Reply by Gekctek** [reply, marked as answer]
A potentially easy fix is to differentiate the routes with a prefix like
```cs
.Map("/JsonRpc", b =>
{
	b.UseJsonRpc();
})
.Map("/Other", b =>
{
       b.UseOther();
});
```

**Reply by eric-davis** [reply]
I created a simple example solution to illustrate what I am seeing: https://github.com/eric-davis/JsonRpc.Router.Test

Right now, if I hit the `BarController` endpoint @ `https://localhost:5001/bar` everything works as expected and I get a valid JsonRpc response back. But if I hit the `FooController` endpoint @ `https://localhost:5001/foo/99`, I get a -32600 error (Json request was empty).

If I comment out the JsonRpc method calls in `Program.cs`, the Foo controller endpoint works as it's supposed to.

**Reply by eric-davis** [reply]
@Gekctek - Your suggestion about the route prefix mapping appears to work. Thanks!

```c#
app.Map("/jsonrpc", options =>
{
    options.UseJsonRpc();
});

app.UseMvc();

```

**Reply by Gekctek** [reply]
I was able to modify your test project to work. Im not sure how the `.MapController()` on the 'WebApplication` object works, I havent used that middleware pipeline before, but here is an updated Program.cs using Routing endpoints for your API that works without mapping:

```cs
using SampleApi.Controllers;

var builder = WebApplication.CreateBuilder(args);

builder.Services.AddControllers();
builder.Services.AddRouting();
builder.Services.AddJsonRpc();

var app = builder.Build();

app.UseHttpsRedirection();

app.UseAuthorization();

app
    .UseRouting()
    .UseEndpoints(b =>
    {
        b.MapControllers();
    })
    .UseJsonRpc(options =>
    {
        options.AddController<BarController>();
    });

app.Run();
```
